cli/pack/common.go

```diff
@@ -285,6 +285,17 @@ func copyApplications(bundleEnvPath string, packCtx *PackCtx,
 		}
 		inst := instances[0]
 		appPath := inst.AppDir
+		if !inst.IsFileApp {
+			ignorePatterns, err := readPackIgnore(appPath)
+			if err != nil {
+				return fmt.Errorf("failed to read .packignore: %v", err)
+			}
+
+			if err = removeIgnoredFiles(appPath, ignorePatterns); err != nil {
```

Comment:
Looks like we remove files before copying the application. This approach will surprise the user.
We should skip copying such files, not removing them from the source directory.